src/kernel/time.zig

```diff
@@ -0,0 +1,31 @@
+const builtin = @import("builtin");
+
+pub const DateTime = struct {
+    second: u32,
+    minute: u32,
+    hour: u32,
+    day: u32,
+    month: u32,
+    year: u32,
+    century: u32,
+    day_of_week: u32,
+};
+
+pub fn getDateTime() DateTime {
+    return switch (builtin.os.tag) {
```

Comment:
I'm a little confused as to why this is necessary, when would we not be in freestanding and why do we need to accommodate for other cases?

Reply:
I use this driver in the build script so runs on the hosts OS, therefore need a non-freestanding version but there isn't one in the standard library yet, see: https://github.com/ziglang/zig/issues/8396

Reply:
I see, but why use this when building? Is it just because it has a nicer interface than what is currently in std? Using kernel code on the host is a little weird to me.